source/common/upstream/upstream_impl.cc

```diff
@@ -317,11 +327,12 @@ HostSetImpl::partitionHosts(HostVectorConstSharedPtr hosts,
                            std::move(degraded_hosts), std::move(degraded_hosts_per_locality));
 }
 
-double HostSetImpl::effectiveLocalityWeight(uint32_t index) const {
+double HostSetImpl::effectiveLocalityWeight(uint32_t index,
+                                            const HostsPerLocality& eligible_hosts) const {
   ASSERT(locality_weights_ != nullptr);
   ASSERT(hosts_per_locality_ != nullptr);
   const auto& locality_hosts = hosts_per_locality_->get()[index];
-  const auto& locality_healthy_hosts = healthy_hosts_per_locality_->get()[index];
+  const auto& locality_healthy_hosts = eligible_hosts.get()[index];
```

Comment:
locality_healthy_hosts -> locality_eligible_hosts?

Should we update comments in general swapping health for eligibility?  locality_availability_ratio?  I guess it's not terrible to say the healthiness ration is based on percent eligible hosts so we can also leave as-is.

Reply:
I like using availability instead, I think it makes more sense now that we're extending this beyond just health to degraded